x/axelarcork/types/proposal.go

```diff
@@ -0,0 +1,240 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	ProposalTypeAddManagedCellarIDs = "AddAxelarManagedCellarIDs"
+
+	ProposalTypeRemoveManagedCellarIDs   = "RemoveAxelarManagedCellarIDs"
+	ProposalTypeScheduledCork            = "AxelarScheduledCork"
+	ProposalTypeCommunitySpend           = "AxelarCommunitySpend"
+	ProposalTypeAddChainConfiguration    = "AddAxelarChainConfiguration"
+	ProposalTypeRemoveChainConfiguration = "RemoveAxelarChainConfiguration"
+)
+
+var _ govtypes.Content = &AddAxelarManagedCellarIDsProposal{}
+var _ govtypes.Content = &RemoveAxelarManagedCellarIDsProposal{}
+var _ govtypes.Content = &AxelarScheduledCorkProposal{}
+var _ govtypes.Content = &AxelarCommunityPoolSpendProposal{}
+var _ govtypes.Content = &AddChainConfigurationProposal{}
+var _ govtypes.Content = &RemoveChainConfigurationProposal{}
+
+func init() {
+	govtypes.RegisterProposalType(ProposalTypeAddManagedCellarIDs)
+	govtypes.RegisterProposalTypeCodec(&AddAxelarManagedCellarIDsProposal{}, "sommelier/AddAxelarManagedCellarIDsProposal")
+
+	govtypes.RegisterProposalType(ProposalTypeRemoveManagedCellarIDs)
+	govtypes.RegisterProposalTypeCodec(&RemoveAxelarManagedCellarIDsProposal{}, "sommelier/RemoveAxelarManagedCellarIDsProposal")
+
+	govtypes.RegisterProposalType(ProposalTypeScheduledCork)
+	govtypes.RegisterProposalTypeCodec(&AxelarScheduledCorkProposal{}, "sommelier/AxelarScheduledCorkProposal")
+
+	govtypes.RegisterProposalType(ProposalTypeAddChainConfiguration)
+	govtypes.RegisterProposalTypeCodec(&AddChainConfigurationProposal{}, "sommelier/AddAxelarChainConfigurationProposal")
+
+	govtypes.RegisterProposalType(ProposalTypeRemoveChainConfiguration)
+	govtypes.RegisterProposalTypeCodec(&RemoveChainConfigurationProposal{}, "sommelier/RemoveAxelarChainConfigurationProposal")
+
+	govtypes.RegisterProposalType(ProposalTypeCommunitySpend)
+	govtypes.RegisterProposalTypeCodec(&AxelarCommunityPoolSpendProposal{}, "sommelier/AxelarCommunitySpendProposal")
+}
+
+func NewAddManagedCellarIDsProposal(title string, description string, chainName string, chainID uint64, cellarIds *CellarIDSet) *AddAxelarManagedCellarIDsProposal {
```

Comment:
We probably want these function names to match the type it's creating, re: the "Axelar" string.

Reply:
added